src/mercury_engine_data_structures/formats/brfld.py

```diff
@@ -11,66 +13,165 @@
+        returns: each actor in the actor layer"""
+        for sublayer_name, sublayer in self.raw.Root.pScenario[actor_layer.value].dctSublayers.items():
             for actor_name, actor in sublayer.dctActors.items():
-                yield layer_name, actor_name, actor
+                yield sublayer_name, actor_name, actor
 
-    def follow_link(self, link: str):
+    def follow_link(self, link: str) -> Any | None:
+        """
+        Gets the object a link is referencing
+
+        param link: the link to follow
+        returns: the part of the BRFLD link is referencing"""
         if link != "{EMPTY}":
             result = self.raw
             for part in link.split(":"):
                 result = result[part]
             return result
 
-    def link_for_actor(self, actor_name: str, layer_name: str = "default") -> str:
-        return ":".join(["Root", "pScenario", "rEntitiesLayer", "dctSublayers", layer_name, "dctActors", actor_name])
+    def link_for_actor(
+        self, actor_name: str, sublayer_name: str = "default", actor_layer: ActorLayer = ActorLayer.ENTITIES
+    ) -> str:
+        """
+        Builds a link for an actor
 
-    def all_actor_groups(self) -> Iterator[str]:
-        yield from self.raw.Root.pScenario.rEntitiesLayer.dctActorGroups.keys()
+        param actor_name: the name of the actor
+        sublayer_name: the name of the sublayer the actor is in
+        actor_layer: the actor layer the actor is in
+        returns: a string representing where in the BRFLD the actor is"""
+        return ":".join(
+            ["Root", "pScenario", actor_layer.value, "dctSublayers", sublayer_name, "dctActors", actor_name]
+        )
 
-    def get_actor_group(self, group_name: str) -> list[str]:
-        return self.raw.Root.pScenario.rEntitiesLayer.dctActorGroups[group_name]
+    def actor_groups_for_actor_layer(self, actor_layer: ActorLayer = ActorLayer.ENTITIES) -> Iterator[str]:
+        """
+        Iterably gets every actor group in an actor layer
 
-    def is_actor_in_group(self, group_name: str, actor_name: str, layer_name: str = "default") -> bool:
-        return self.link_for_actor(actor_name, layer_name) in self.get_actor_group(group_name)
+        param actor_layer: the actor layer to get the actor groups of
+        returns: each actor group in the actor layer"""
+        yield from self.raw.Root.pScenario[actor_layer.value].dctActorGroups.keys()
 
-    def add_actor_to_group(self, group_name: str, actor_name: str, layer_name: str = "default"):
-        group = self.get_actor_group(group_name)
-        actor_link = self.link_for_actor(actor_name, layer_name)
+    def get_actor_group(self, group_name: str, actor_layer: ActorLayer = ActorLayer.ENTITIES) -> list[str]:
+        """
+        Gets an actor group
+
+        param group_name: the name of the actor group
+        param actor_layer: the actor layer the actor group is in
+        returns: a list of links to actors"""
```

Comment:
while we're in here doing this, I might suggest adding a type alias like `ScenarioLink = str` and using that in our signatures (e.g. making the return type of this function `list[ScenarioLink]` instead of `list[str]`) for more clear semantics